components/src/dynamo/thompson_router/router.py

```
@@ -0,0 +1,734 @@
+        """
+        loads = await self.kv_router.get_potential_loads(token_ids)
+        native_pick, _, _ = await self.kv_router.best_worker(token_ids)
+
+        # Get clean per-request overlap from KvIndexer if available
+        indexer_overlap = None
+        if self.kv_indexer is not None:
+            try:
+                indexer_overlap = await self.kv_indexer.find_matches_for_request(
+                    token_ids
+                )
+            except Exception:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)